blog/content/post/2020-04-27-trenchboot-nlnet-release-04.md

```diff
@@ -0,0 +1,290 @@
+      building Nix...
+      building the system configuration...
+      ```
+
+      5. Reboot platform
+
+      ```
+      $ reboot
+      ```
+
+
+
+#### Requirements verification - LZ content and layout
+
+There are two requirements which LZ must met:
+
+- bootloader information header in LZ should be moved at the end of the LZ
+binary or the end of code section of the LZ;
+- the size of the measured part of the SLB must be set to the code size only;
+
+Bootloader information header is special data structure which content is
+hardcoded in [landing-zone source code](https://github.com/TrenchBoot/landing-zone/blob/master/include/boot.h#L50).
+The requirement is to keep that data in `lz_header.bin` file after code section.
+Hence, when Landing Zone makes measurements, only code section is measured.
+Verification of above requirements can be carried out like this:
+
+      1. Check the value of second word of `lz_header.bin` file using `hexdump`
+      tool.
+
+      ```
+      $ hexdump -C lz_header.bin | head                                                    
+      00000000  d4 01 00 d0 00 00 00 00  00 00 00 00 00 00 00 00  |................|
+      00000010  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
+      *
+      000001d0  00 00 00 00 89 c5 8d a5  00 02 00 00 b9 14 01 01  |................|
+      000001e0  c0 0f 32 83 e0 f9 0f 30  01 ad bc 27 00 00 01 ad  |..2....0...'....|
+      000001f0  52 02 00 00 01 ad 00 90  00 00 01 ad 00 80 00 00  |R...............|
+      00000200  01 ad 08 80 00 00 01 ad  10 80 00 00 01 ad 18 80  |................|
+      00000210  00 00 01 ad 00 40 00 00  0f 01 95 ba 27 00 00 b8  |.....@......'...|
+      00000220  10 00 00 00 8e d8 8e c0  0f 20 e1 83 c9 20 0f 22  |......... ... ."|
+      00000230  e1 8d 85 00 90 00 00 0f  22 d8 b9 80 00 00 c0 0f  |........".......|
+      ```
+
+      > Value of second word (`00 d0`) is size of code section and concurrently
+      offset (address) of bootloader information header. Used notation is
+      little-endian so the value is actually `0xd000` (NOT 0x00d0).
+
+      2. Check the content of `lz_header.bin` file from `0xd000` address.
+
+      ```
+      $ hexdump -C -s 0xd000 lz_header.bin | head                                          
+      0000d000  78 f1 26 8e 04 92 11 e9  83 2a c8 5b 76 c4 cc 02  |x.&......*.[v...|
+      0000d010  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
+      *
+      0000d030  04 00 00 00 10 00 00 00  05 00 00 00 47 4e 55 00  |............GNU.|
+      0000d040  01 00 00 c0 04 00 00 00  01 00 00 00 00 00 00 00  |................|
+      0000d050
+      ```
+
+      > As you can see, there is `78 f1 26 8e 04 92 11 e9  83 2a c8 5b 76 c4 cc
+      02` data under 0xd000 address.
+
+      3. If above data is the same as in data structure in [landing-zone source
+      code](https://github.com/TrenchBoot/landing-zone/blob/master/include/boot.h#L50)
```

Comment:
I believe this must be in one line, at least GitHub has problems rendering it: https://github.com/3mdeb/news-and-ideas/blob/ef4449f8695e8afa7e0d0a81ac742778fba00988/blog/content/post/2020-04-27-trenchboot-nlnet-release-04.md

Reply:
Fixed